torch/_dynamo/variables/builder.py

```diff
@@ -1167,25 +1167,33 @@ def wrap_to_fake_tensor_and_record(
             dynamic_dims = []
             constraint_dims = []
             for i in range(e.dim()):
+                # NB: mark dynamic has precedence over static
+                marked_dynamic = i in getattr(e, "_dynamo_dynamic_indices", set())
+                marked_static = i in getattr(e, "_dynamo_static_indices", set())
```

Comment:
If it has both, should we just raise? Feels like user error, and it would be irritating to think you are marking something static and not seeing anything change...

Reply:
I'm slightly in favor of defining a precedence instead, mostly because I had some experiences with the mark_dynamic having too aggressive error checking that I needed to relax.